src/routes/lstm_routes.py

```diff
@@ -65,51 +65,31 @@ def predict():
-    if error:
-        error_message, status_code = error
-        return jsonify(error_message), status_code
-
-    if validated_data['isCompleted']:
-        previous_sessions = get_all_sessions_from_db(validated_data['deviceId'])
-        try:
-            save_session_to_db(validated_data)
-        except Exception as e:
-            logging.error('An error occurred while saving a session: %s', e)
-            return jsonify({'error': 'An error occurred while saving a session.'}), 500
-        if not previous_sessions:
-            return jsonify({'message': 'Session saved, but Model for device ' + validated_data[
-                'deviceId'] + ' was not trained as no previous session was found.'})
-
-        combined_session_periods = validated_data['sessionPeriods'].copy()
-        previous_sessions.reverse()
-        # Iterate over the reversed previous_sessions and add periods until batch_size is reached
-        for session in previous_sessions:
-            if len(combined_session_periods) < meditation_lstm.TRAINING_DATA_ARR_SIZE:
-                additional_periods_needed = meditation_lstm.TRAINING_DATA_ARR_SIZE - len(combined_session_periods)
-                session_periods = session.to_dict()['sessionPeriods']
-                # Take only as many periods as needed from this session
-                combined_session_periods.extend(session_periods[:additional_periods_needed])
-            else:
-                break
-
-        training_data_arr = map_session_periods_to_training_data(combined_session_periods)
-        logging.info("Length of training_data_arr: " + str(len(training_data_arr)))
-    else:
+    device_id: str = request.args.get('deviceId')
+    if device_id is None or len(device_id) == 0:
+        return jsonify({'error': 'No device id provided.'}), 400
+    print(f"Request on /train_model received from device with id: {device_id}")
+    previous_sessions = get_all_sessions_from_db(device_id)
+    if not previous_sessions:
+        return jsonify({'message': f'No previous sessions found for device {device_id}. Could not train model.'})
+    previous_sessions.reverse()
+    session_periods: list = []
+    # Iterate over the reversed previous_sessions and add periods until batch_size is reached
+    for session in previous_sessions:
+        if len(session_periods) < meditation_lstm.TRAINING_DATA_ARR_SIZE:
+            session_periods.extend(session.to_dict()['sessionPeriods'])
+        else:
+            break
+    if len(session_periods) < meditation_lstm.TRAINING_DATA_ARR_SIZE:
         return jsonify(
-            {'message': 'Model for device ' + validated_data['deviceId'] + ' not trained. Session did not complete.'})
-
+            {'message': f'Model for device "{device_id}" not trained. Could not find enough previous sessions.'})
+    training_data_arr = map_session_periods_to_training_data(session_periods[:meditation_lstm.TRAINING_DATA_ARR_SIZE])
```

Comment:
I think the it's not quite correct how we compose `session_periods`. And it apparently also was incorrect previously. But now if I understand it right, then we might actually cut off most recent data, which is really not desirable for training the model.

We reverse `previous_sessions`. Then we should actually also reverse `session_periods` and `heartRateMeasurements` of one period. Seems like quite an overhead to me.

Instead we might reverse only `previous_sessions`, then insert the respective session periods of one session to our variable `session_periods` at index 0, so that older information is at the start of the array. Do that until at some point `len(session_periods) >= meditation_lstm.TRAINING_DATA_ARR_SIZE`. Then cut the rest off like we do here.

I hope that's somewhat understandable.